Drop-In/src/androidTest/java/com/braintreepayments/api/AddCardFragmentUITest.kt

```diff
@@ -0,0 +1,137 @@
+
+            assertNull(fragment.cardForm.cardEditText.textInputLayoutParent?.error)
+        }
+    }
+
+    @Test
+    fun whenStateIsRESUMED_sendsAnalyticsEventToNotifyAddCardSelected() {
+        // TODO: assert 'card.selected' analytics event sent
+    }
+
+    @Test
+    fun whenStateIsRESUMED_onSubmitButtonClick_showsLoader() {
+        val scenario = FragmentScenario.launchInContainer(AddCardFragment::class.java, null, R.style.bt_drop_in_activity_theme)
+        scenario.moveToState(Lifecycle.State.RESUMED)
+
+        scenario.onFragment { fragment ->
+            fragment.dropInViewModel.setSupportedCardTypes(listOf(CardType.VISA))
+        }
+
+        onView(isRoot()).perform(waitFor(500))
+        onView(withId(R.id.bt_card_form_card_number)).perform(typeText(VISA))
+        onView(withId(R.id.bt_button)).perform(click())
+        onView(withId(R.id.bt_button)).check(matches(not(isDisplayed())))
+    }
+
+    @Test
+    fun whenStateIsRESUMED_onSubmitButtonClickWithValidCardNumber_sendsAddCardEvent() {
+        val scenario = FragmentScenario.launchInContainer(AddCardFragment::class.java, null, R.style.bt_drop_in_activity_theme)
+        scenario.moveToState(Lifecycle.State.RESUMED)
+
+        scenario.onFragment { fragment ->
+            fragment.dropInViewModel.setSupportedCardTypes(listOf(CardType.VISA))
+        }
+
+        onView(isRoot()).perform(waitFor(500))
+        onView(withId(R.id.bt_card_form_card_number)).perform(typeText(VISA))
+        onView(withId(R.id.bt_button)).perform(click())
+
+        scenario.onFragment { fragment ->
+            val activity = fragment.requireActivity()
+            val fragmentManager = fragment.parentFragmentManager
+            fragmentManager.setFragmentResultListener("BRAINTREE_EVENT", activity) { requestKey, result ->
+                val event = result.get("BRAINTREE_RESULT") as AddCardEvent
+                assertEquals(VISA, event.cardNumber)
+                countDownLatch.countDown()
+            }
+        }
+        countDownLatch.await()
+    }
+
+    @Test
+    fun whenStateIsRESUMED_whenCardNumberValidationErrorsArePresentInViewModel_displaysErrorsInlineToUser() {
+        val scenario = FragmentScenario.launchInContainer(AddCardFragment::class.java, null, R.style.bt_drop_in_activity_theme)
+        scenario.moveToState(Lifecycle.State.RESUMED)
+
+        scenario.onFragment { fragment ->
+            fragment.dropInViewModel.setSupportedCardTypes(listOf(CardType.VISA))
+            fragment.dropInViewModel.setCardTokenizationError(ErrorWithResponse.fromJson(Fixtures.CREDIT_CARD_ERROR_RESPONSE))
+
+            assertEquals(fragment.context?.getString(R.string.bt_card_number_invalid),
+                    fragment.cardForm.cardEditText.textInputLayoutParent?.error)
+        }
+    }
+
+    @Test
+    fun whenStateIsRESUMED_whenCardFromInvalid_showsButton() {
```

Comment:
```suggestion
    fun whenStateIsRESUMED_whenCardFormInvalid_showsButton() {
```

Reply:
This test is a little confusing - which button are we looking for?

Reply:
I think that would be the submit button here yeah, we should clarify.